Stripe/STPRedirectContext.m

```diff
@@ -138,18 +149,21 @@ - (void)subscribeToUrlAndForegroundNotifications {
 }
 
 - (void)unsubscribeFromNotificationsAndDismissPresentedViewControllers {
+    [self unsubscribeFromNotifications];
+    [self dimissPresentedViewController];
+}
+
+- (void)unsubscribeFromNotifications {
     [[NSNotificationCenter defaultCenter] removeObserver:self
                                                     name:UIApplicationWillEnterForegroundNotification
                                                   object:nil];
     [[STPURLCallbackHandler shared] unregisterListener:self];
+}
 
+- (void)dimissPresentedViewController {
     if (self.safariVC) {
         [self.safariVC.presentingViewController dismissViewControllerAnimated:YES
-                                                                   completion:^{
-                                                                       stpDispatchToMainThreadIfNecessary(^{
-                                                                           [self handleRedirectCompletionWithError:nil];
-                                                                       });
-                                                                   }];
+                                                                   completion:nil];
```

Comment:
so we don't need the `completion` anymore because we're *sure* that one of the delegates will be called?

maybe leave a comment so we don't regress on this.